### PR TITLE
Added connection to pull data from apple healthkit

### PR DESCRIPTION
**Pull request by cara-wong**

### CODE REVIEW

**Comment by cara-wong** [comment]
This PR sets up the connection to HealthKit, and includes code to render gender on the home screen once you've logged in to test the connection. 

Steps to test: 

1. You must have pulled down the master branch and set up the front and back end of the app so that you are able to login.
2. Pull down just the changed files from this PR to your project:
	mobile/ios/CompanionKit/Info.plist
	mobile/package.json
	mobile/src/hooks/useHealthKit.tsx
	mobile/src/screens/components/TipCard.tsx
3. From within the mobile dir:
    - Run _yarn install_ and _react-native link rn-apple-healthkit_  
    - Run _cd ./ios && pod install_
4. Open the project in Xcode and enable HealthKit: https://developer.apple.com/documentation/healthkit/setting_up_healthkit
5. Build and run the app from Xcode on a real device. You should be prompted to enable permissions when the app loads, and when you log in you will see your gender on the tip card.

HealthKit-Test

**Comment by rlnsy** [comment]
@cara-wong Just got this to work in the simulator!

File: mobile/src/hooks/useHealthKit.tsx
Comment (rlnsy) [inline mobile/src/hooks/useHealthKit.tsx:6]: Should we abstract this into something more general like `getHealthData`? So that we have a more general purpose util before merging in

File: mobile/src/hooks/useHealthKit.tsx
Comment (rlnsy) [inline mobile/src/hooks/useHealthKit.tsx:26]: can we change this so that gender is a single field in data?

File: mobile/src/screens/components/TipCard.tsx
Comment (rlnsy) [inline mobile/src/screens/components/TipCard.tsx:67]: We should guard against errors and undefined data here

**Review by rlnsy — CHANGES_REQUESTED** [review]

**Review by crugwiro — APPROVED** [review]

**Comment by cara-wong** [comment]
Steps to Build App on Real Device with Xcode:

1. In Navigation at the top of Xcode: Xcode>Preferences>Accounts: sign in with your apple ID
2. In folder navigation select “CompanionKit”, and in the tab that opens select “Signing & Capabilities”. Set team to personal team.
3. Open “Keychain Access” from your search bar and allow access to your apple development key.
4. Once you build the app, you will need to go into your phone’s settings: Settings>General>Profiles & Device Management and trust your Apple Development account

@rlnsy

File: mobile/src/hooks/useHealthKit.tsx
Comment (cara-wong) [reply mobile/src/hooks/useHealthKit.tsx]: done - see new commits

File: mobile/src/hooks/useHealthKit.tsx
Comment (cara-wong) [reply mobile/src/hooks/useHealthKit.tsx]: done - see new commits

File: mobile/src/screens/components/TipCard.tsx
Comment (cara-wong) [reply mobile/src/screens/components/TipCard.tsx]: Right now it's protecting against loading errors and if the gender is not set in healthkit, "unknown" will be displayed. I'm thinking we should wait on adding more guards until we have more information on our use cases (what exact data we want to retrieve and how we'll use the data).

Adding this code in the TipCard was just to test the connection was working. I'm going to remove it before it's merged. Let me know what you think.

Comment (rlnsy) [reply mobile/src/screens/components/TipCard.tsx]: Yup! - Wanted to confirm that unknown was the catch-all case, and not a specific value. Sounds good to me

**Review by rlnsy — APPROVED** [review]

**Comment by wynnset** [comment]
@cara-wong will update this PR and move any info to Notion before marking it as ready for merge

**Comment by cara-wong** [comment]
@wynnset Removed test code and added testing instructions from the comments to Notion. Ready to merge.

**Comment by wynnset** [comment]
I think I'll hold off on merging this until we actually build this into the app since it is only partial functionality and currently doesn't actually record anything.

**Comment by cara-wong** [comment]
@wynnset Resolved merge conflicts - ready to merge when you think it makes sense to

**Pull request merged**
